### ISSUE TITLE
Maybe types in fields when using fragment spread are wrong.

### ISSUE THREAD

**Issue by Fhoerth** [dotansimha/graphql-code-generator#2527]
**Description**
Maybe types when spreading fragments aren't consistent with the original type.

**To Reproduce**

1. My GraphQL schema:

```graphql
type User {
  username: String
}


type Order {
  id: String!
  user: User
}


type Query {
  order(id: String!): Order
}


schema {
  query: Query
}

```

2. My GraphQL operations:

```graphql
fragment UserFields on User {
  username
}

fragment OrderFields on Order {
  id
  user {
    ...UserFields
  }
}

query getOrder($id: String!) {
  order(id: $id) {
    ...OrderFields
  }
}
```

3. My `codegen.yml` config file:

```yml
generates:
  types.flow.js:
    - flow
    - flow-operations
```

**Expected behavior**
Note the `?` after `user`.
```javascript
export type OrderFieldsFragment = (
  { __typename?: 'Order' }
  & $Pick<Order, { id: * }>
  & { user: ?({ __typename?: 'User' }
    & UserFieldsFragment
   })
);
```

For example, if a component accepts the prop `orderFields: OrderFieldsFragment `, then flow won't throw any warnings about `user.username` while it should warn `Cannot get orderFields.user.username because property id is missing in undefined [1].`

I can confirm this works OK in the `1.4.0` version.
You can test it in https://transform.tools/graphql-to-flow (this page uses `^1.4.0` pkg version)

The generated code from the `1.4.0` version is:
```javascript
export type OrderFieldsFragment = { __typename?: "Order" } & $Pick<
  Order,
  { id: * }
> & { user: ?({ __typename?: "User" } & UserFieldsFragment) };
```

**Comment by dotansimha** [comment]
I think this was fixed in: https://github.com/dotansimha/graphql-code-generator/pull/2540 
Can you please try the latest alpha version? `1.7.1-alpha-7846646a.59`

**Comment by dotansimha** [comment]
Fixed in 1.8.0 🚀
